lib/Serialization/Deserialization.cpp

```diff
@@ -2424,7 +2425,29 @@ ModuleFile::resolveCrossReference(ModuleID MID, uint32_t pathLen) {
         break;
       }
 
-      if (!privateDiscriminator.empty()) {
+      if (importedFromClang && !privateDiscriminator.empty()) {
```

Comment:
I wonder if it would be better collapse this into the second case with a subsequent check for just `importedFromClang`, i.e.:

```c++
if (!privateDiscriminator.empty()) {
  if (importedFromClang) {
  } else {
  }
}
```